File: Seminar/src/main/java/org/sopt/Seminar/domain/user/repository/UserRepository.java
package org.sopt.Seminar.domain.user.repository;

import java.util.Optional;
import org.sopt.Seminar.domain.user.model.User;
import org.springframework.data.jpa.repository.JpaRepository;

public interface UserRepository extends JpaRepository<User, Long> {
    Optional<User> findByNickname(String nickname);
}

File: kopring/week2/kotlin-semina/src/main/kotlin/org/sopt/kotlinsemina/domain/user/repository/UserRepository.kt
package org.sopt.kotlinsemina.domain.user.repository

import org.sopt.kotlinsemina.domain.user.model.User
import org.springframework.data.jpa.repository.JpaRepository

interface UserRepository: JpaRepository<User, Long> {
    fun findByNickname(nickname: String): User?
}